quote-book/sqlite/migrations/2023-02-17-204631_quotes/up.sql

```diff
@@ -0,0 +1,48 @@
+-- Note about primary keys: For SQLite to auto-increment a PRIMARY KEY row, the column must be `INTEGER PRIMARY KEY`.
+-- Even `INT PRIMARY KEY` won't work. It has to be `INTEGER PRIMARY KEY`. See https://www.sqlite.org/autoinc.html.
+-- This has an annoying impact on Diesel - it forces the primary key column to be `i32` even though SQLite uses 64 bit for these columns.
+-- Diesel also requires that a table has a primary key column, so each table must have one.
+--
+-- Note about signed/unsigned: SQLite does not support unsigned types. As such, everything is signed and gets represented
+-- as i32/i64 on the rust side. Rust code can still safely cast to unsigned types without losing data, but this does mean
+-- that SQLite functions will behave incorrectly - for example calling SUM(amount) should be avoided, since SQLite will be
+-- summing signed values. The workaround is to do such operations in Rust, after casting to the unsigned type.
+-- Where we want to do sorting/range comparisons (e.g. greater-than, less-than, etc) on u64s, we revert to storing
+-- them as the big endian bytes. SQLite uses memcmp() to compare binary values, so this works.
+
+CREATE TABLE quotes (
+    -- Quote ID
+    id BINARY PRIMARY KEY NOT NULL,
+
+    -- The SCI, protobuf-encoded
+    sci_protobuf BINARY NOT NULL,
+
+    -- Base token id (stored as signed i64 since we can cast from rust and only do equality comparisons on it)
+    base_token_id BIGINT NOT NULL,
+
+    -- Counter token id (stored as signed i64 since we can cast from rust and only do equality comparisons on it)
+    counter_token_id BIGINT NOT NULL,
+
+    -- The range of base tokens offered by this quote. Stored as big endian bytes so that we don't run into signed/unsigned issues.
+    -- These are the raw values from the quote.
+    base_range_min BINARY NOT NULL,
+    base_range_max BINARY NOT NULL,
+
+    -- The number of counter tokens needed to trade the max amount of base tokens.
+    max_counter_tokens BINARY NOT NULL,
+
+    -- Timestamp at which the quote arrived
+    timestamp BIGINT NOT NULL,
+
+    -- The key image of the quote.
+    key_image BINARY NOT NULL,
+
+    -- The tombstone block of the quote
```

Comment:
Don't we do inequality comparisons with the tombstone block? Should this also be in Binary for the comparisons to work for unsigned?

Reply:
i think if we actually have more than `i64::Max` blocks in the mobilecoin blockchain, we are all going to be either sipping margaritas on a beach somewhere, or else, dead 500 years ago

Reply:
but like, you're not wrong, we could make it binary i guess

Reply:
yeah i dont think we need to worry about exceeding i64 block count